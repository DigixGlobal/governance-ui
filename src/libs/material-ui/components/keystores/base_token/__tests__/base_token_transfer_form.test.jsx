import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import BaseTokenTransferForm from '../base_token_transfer_form';

const web3 = {
  eth: {
    getBalance: jest.fn,
    balance: () => ({
      minus: () => ({
        shift: () => ({}),
      }),
    }),
  },
};

const network = {
  symbol: 'GBP',
  color: 'blue',
};

const formData = {
  from: '0x1234',
  gas: 1000,
  gasPrice: 10,
};

describe('<BaseTokenTransferForm />', () => {
  test('renders correctly', () => {
    const component = shallow(
      <BaseTokenTransferForm
        formChange={jest.fn}
        formData={formData}
        web3={web3}
        network={network}
      />,
    );

    expect(toJson(component)).toMatchSnapshot();
  });
});
