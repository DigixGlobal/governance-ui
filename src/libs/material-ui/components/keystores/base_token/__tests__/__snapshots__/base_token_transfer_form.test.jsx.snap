// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BaseTokenTransferForm /> renders correctly 1`] = `
<FormField>
  <AddressInput
    formChange={[Function]}
    formData={
      Object {
        "from": "0x1234",
        "gas": 1000,
        "gasPrice": 10,
      }
    }
    hideDropdown={false}
    label="From"
    name="from"
    placeholder="e.g. \`0x123...456\`"
    showQrScanner={false}
  />
  <AddressInput
    formChange={[Function]}
    formData={
      Object {
        "from": "0x1234",
        "gas": 1000,
        "gasPrice": 10,
      }
    }
    hideDropdown={false}
    label="To"
    name="to"
    placeholder="e.g. \`0x123...456\`"
    showQrScanner={true}
  />
  <ValueInput
    color="blue"
    formChange={[Function]}
    formData={
      Object {
        "from": "0x1234",
        "gas": 1000,
        "gasPrice": 10,
      }
    }
    label="Value"
    name="ethValue"
    sendAll={Object {}}
    symbol="GBP"
  />
</FormField>
`;
